addon/Feeds/MetadataFeed.js

```diff
@@ -54,28 +57,37 @@ module.exports = class MetadataFeed extends Feed {
   getData() {
     const experiments = this.store.getState().Experiments.values;
 
-    let links = Array.from(this.linksToFetch.keys(), item => Object.assign({"url": item}));
-    this.linksToFetch.clear();
+    const links = this.urlsToFetch.map(url => ({"url": url}));
+    this.urlsToFetch = [];
 
     // Make no requests for metadata
-    if (experiments.metadataNoService) {
+    if (experiments.metadataNoService || experiments.metadataLocalRefresh) {
       return this.options.fetchNewMetadataLocally(links, "METADATA_FEED_REQUEST").then(() => (am.actions.Response("METADATA_UPDATED")));
     }
 
     return this.options.fetchNewMetadata(links, "METADATA_FEED_REQUEST").then(() => (am.actions.Response("METADATA_UPDATED")));
   }
   onAction(state, action) {
+    const experiments = this.store.getState().Experiments.values;
+
     switch (action.type) {
       case am.type("APP_INIT"):
         this.getInitialMetadata("app was initializing");
         break;
       case am.type("RECEIVE_PLACES_CHANGES"):
-        this.linksToFetch.set(action.data.url, Date.now());
-        if (this.linksToFetch.size > MAX_NUM_LINKS) {
+        this.urlsToFetch.push(action.data.url);
+
+        if (experiments.metadataLocalRefresh && ((Date.now() - this.lastRefreshed) >= UPDATE_TIME)) {
+          this.getInitialMetadata("periodically refreshing metadata");
+        }
+
+        if (this.urlsToFetch.length > MAX_NUM_LINKS) {
```

Comment:
does the order of these `if` statements matter? I can see the `getInitialMetadata` above would cause this refresh below to be way down at the end of the queue. But it probably doesn't matter for most cases.